var/spack/repos/builtin/packages/gunrock/package.py

```diff
@@ -23,9 +27,110 @@ class Gunrock(CMakePackage, CudaPackage):
     version('0.2',      submodules=True, tag='v0.2')
     version('0.1',      submodules=True, tag='v0.1')
 
-    depends_on('cuda')
+    variant('cuda', default=True, description="Build with Cuda support")
+
+    variant('lib',                  default=True,  description='Build main gunrock library')
+    variant('shared_libs',          default=True,  description='Turn off to build for static libraries')
+    variant('tests',                default=True,  description='Build functional tests / examples')
+    variant('mgpu_tests',           default=False, description='Builds Gunrock applications and enables the ctest framework for single GPU implementations')
+    variant('cuda_verbose_ptxas',   default=False, description='Enable verbose output from the PTXAS assembler')
+    variant('google_tests',         default=False, description='Build unit tests using googletest')
+    variant('code_coverage',        default=False, description="run code coverage on Gunrock's source code")
+    variant('all_applications',     default=True,  description='Build all applications')
+    # apps
+    variant('app_bc',       default=False, description='Only build BC primitive')
+    variant('app_bfs',      default=False, description='Only build BFS primitive')
+    variant('app_cc',       default=False, description='Only build CC primitive')
+    variant('app_pr',       default=False, description='Only build PR primitive')
+    variant('app_sssp',     default=False, description='Only build SSSP primitive')
+    variant('app_dobfs',    default=False, description='Only build DOBFS primitive')
+    variant('app_hits',     default=False, description='Only build HITS primitive')
+    variant('app_salsa',    default=False, description='Only build SALSA primitive')
+    variant('app_mst',      default=False, description='Only build MST primitive')
+    variant('app_wtf',      default=False, description='Only build WTF primitive')
+    variant('app_topk',     default=False, description='Only build TOPK primitive')
```

Comment:
I think this might be worth a multi-valued variant. Maybe something like:
```suggestion
    variant(
        "applications",
        values=disjoint_sets(
            ('all',), ('bc', 'bfs', 'cc', 'pr', 'sssp', ...)
        ).allow_empty_set().with_default('all').with_error(...),
        description="Application to be built"
```
If this works (and it should :slightly_smiling_face: ) it will avoid all the conflicts below.